src/Sylius/Bundle/ApiBundle/Validator/Constraints/ChosenPaymentMethodEligibilityValidator.php

```diff
@@ -39,13 +53,17 @@ public function validate($value, Constraint $constraint): void
 
         $paymentMethod = $this->paymentMethodRepository->findOneBy(['code' => $value->getPaymentMethodCode()]);
 
-        if ($paymentMethod === null) {
-            $this->context->addViolation(
-                $constraint->message,
-                ['%paymentMethodCode%' => $value->getPaymentMethodCode()]
-            );
+        /** @var PaymentInterface $payment */
+        $payment = $this->paymentRepository->find($value->paymentId);
+        Assert::notNull($payment);
 
+        if ($paymentMethod === null) {
+            $this->context->addViolation($constraint->paymenyMethodNotExist, ['%code%' => $value->getPaymentMethodCode()]);
             return;
         }
```

Comment:
```suggestion
     
        if ($paymentMethod === null) {
            $this->context->addViolation($constraint->paymenyMethodNotExist, ['%code%' => $value->getPaymentMethodCode()]);
            return;
        }
        
        /** @var PaymentInterface $payment */
        $payment = $this->paymentRepository->find($value->paymentId);
        Assert::notNull($payment);
```

Sorry if formated wrongly, but i will move the second call to repository after the check. One less call to repository if paymentMethod will not be found is one less call 😄